### PR TITLE
[receiver/azureeventhub] remove continue from timestamp parse failure

### PR DESCRIPTION
**Pull request by nslaughter**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
- Timestamp parse failure drops log record #31244

**Testing:** <Describe what testing was performed and which tests were added.>
Analyzed log records with and without this change.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by mwear — APPROVED** [review]
LGTM

**Comment by cparkins** [comment]
@nslaughter - If this is standard practice elsewhere, I think it makes sense. I intentionally skipped these records because they are really hard to find in Splunk. Maybe it makes sense to add a warning? I'll look at other receivers and see what they do.  What do you think @atoulme?

File: pkg/translator/azure/resourcelogs_to_logs.go
Comment (cparkins) [inline pkg/translator/azure/resourcelogs_to_logs.go:113]: Looking at how this is handled elsewhere, I think we should not set the timestamp in this scenario and set the observed timestamp to the current time in both scenarios.

**Review by cparkins — CHANGES_REQUESTED** [review]

File: pkg/translator/azure/resourcelogs_to_logs.go
Comment (nslaughter) [reply pkg/translator/azure/resourcelogs_to_logs.go:113]: > Looking at how this is handled elsewhere, I think we should not set the timestamp in this scenario and set the observed timestamp to the current time in both scenarios.

Good call. I'm posting the description of the field for documentation:

> Field: ObservedTimestamp
>
> Type: Timestamp, uint64 nanoseconds since Unix epoch.
> 
> Description: Time when the event was observed by the collection system. For events that originate in OpenTelemetry (e.g. using OpenTelemetry Logging SDK) this timestamp is typically set at the generation time and is equal to Timestamp. For events originating externally and collected by OpenTelemetry (e.g. using Collector) this is the time when OpenTelemetry’s code observed the event measured by the clock of the OpenTelemetry code. This field SHOULD be set once the event is observed by OpenTelemetry.
> 
> For converting OpenTelemetry log data to formats that support only one timestamp or when receiving OpenTelemetry log data by recipients that support only one timestamp internally the following logic is recommended:
> 
> Use Timestamp if it is present, otherwise use ObservedTimestamp.

Comment (cparkins) [reply pkg/translator/azure/resourcelogs_to_logs.go:113]: I think that looks correct based on how it's handled by the standard OTLP conversion and the Documentation. Nice work!

**Review by cparkins — APPROVED** [review]

**Comment by nslaughter** [comment]
hi @dashpole, I have an approving review from a codeowner. However, this may not show it as he wasn't a codeowner when the PR began? ... So I'm pinging. Hoping you can help getting this merged as assignee. Thank you!

File: pkg/translator/azure/resourcelogs_to_logs.go
Comment (dashpole) [inline pkg/translator/azure/resourcelogs_to_logs.go:116]: nit: you can move `lr := logRecords.AppendEmpty()` to before you call `getTimestamp`, and then make the check below in an `} else {` block.

**Review by dashpole — APPROVED** [review]

**Pull request closed without merging**
